File: PassGen.cpp






// *~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~*


#include <iostream>
#include <fstream>
#include <string>
#include <cstdlib>
#include <ctime>
#include <vector>

using namespace std;


string generatePassword(int length) {
    const string characters =
        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()-_=+[]{}|;:',.<>?/";
    string password;
    srand(time(0)); 

    for (int i = 0; i < length; ++i) {
        password += characters[rand() % characters.size()];
    }
    return password;
}


void ensureFileExists(const string& filename) {
    ifstream file(filename);
    if (!file) { 
        ofstream createFile(filename);
        createFile.close();
    }
}


void savePasswordToFile(const string& filename, const string& name, const string& password) {
    ensureFileExists(filename);
    ofstream file(filename, ios::app); // 
    if (file.is_open()) {
        file << name << ": " << password << "\n";
        file.close();
    }
    else {
        cerr << "File is not accessible\n";
    }
}

int main() {
    string name;
    int length;
    const string filename = "passlog.txt";

    
    cout << "Enter the name of password: ";
    getline(cin, name);

    cout << "Enter password length: ";
    cin >> length;

    
    string password = generatePassword(length);
    savePasswordToFile(filename, name, password);

    cout << "\n" << "Password generated and saved in " << filename << ".\n";

    return 0;
}






// *~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~**~*

File: Passgen.cs

using System;
using System.IO;
using System.Text;

class Passgen
{
    static void Main()
    {

        // By changing this, the storage file for passwords will be renamed to the designated name.
        string filename = "passlog.txt";


        Console.Write("Enter password name: ");
        string name = Console.ReadLine();

        Console.Write("Enter password length: ");
        if (!int.TryParse(Console.ReadLine(), out int length) || length <= 0)
        {
            Console.WriteLine("Invalid length");
            return;
        }

        
        EnsureFileExists(filename);

        
        string password = GeneratePassword(length);

        
        SavePasswordToFile(filename, name, password);

        Console.WriteLine( "\n" + $"Password generated and saved in {filename}.");
    }

    static void EnsureFileExists(string filename)
    {
        if (!File.Exists(filename))
        {
            File.Create(filename).Dispose(); 
        }
    }

    static string GeneratePassword(int length)
    {
        const string characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()-_=+[]{}|;:',.<>?/";
        StringBuilder password = new StringBuilder();
        using (var rng = new System.Security.Cryptography.RNGCryptoServiceProvider())
        {
            byte[] buffer = new byte[1];
            while (password.Length < length)
            {
                rng.GetBytes(buffer);
                char character = (char)buffer[0];
                if (characters.IndexOf(character) != -1)
                {
                    password.Append(character);
                }
            }
        }
        return password.ToString();
    }

    static void SavePasswordToFile(string filename, string name, string password)
    {
        using (StreamWriter sw = new StreamWriter(filename, append: true))
        {
            sw.WriteLine($"{name}: {password}");
        }
    }
}



